core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -0,0 +1,99 @@
+package io.grpc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Encloses classes related to the compression and decompression of messages.
+ */
+public final class MessageEncoding {
+
+  /**
+   * Special sentinel codec indicating that no compression should be used.  Users should use
+   * reference equality to see if compression is disabled.
+   */
+  public static final Codec NONE = new Codec() {
+    @Override
+    public InputStream decompress(InputStream is) throws IOException {
+      return is;
+    }
+
+    @Override
+    public String getMessageEncoding() {
+      return null;
+    }
+
+    @Override
+    public OutputStream compress(OutputStream os) throws IOException {
+      return os;
+    }
+  };
+
+  /**
+   * Represents a message compressor.
+   */
+  public interface Compressor {
+    /**
+     * Returns the message encoding that this compressor uses.
+     *
+     * <p>This can be values such as "gzip", "deflate", "snappy", etc.
+     */
+    String getMessageEncoding();
+
+    /**
+     * Wraps an existing output stream with a compressing output stream.
+     * @param os The output stream of uncompressed data
+     * @return An output stream that compresses
+     */
+    OutputStream compress(OutputStream os) throws IOException;
+  }
+
+  /**
+   * Represents a message decompressor.
+   */
+  public interface Decompressor {
+    /**
+     * Returns the message encoding that this compressor uses.
+     *
+     * <p>This can be values such as "gzip", "deflate", "snappy", etc.
+     */
+    String getMessageEncoding();
+
+    /**
+     * Wraps an existing input stream with a decompressing input stream.
+     * @param is The input stream of uncompressed data
+     * @return An input stream that decompresses
+     */
+    InputStream decompress(InputStream is) throws IOException;
+  }
+
+  /**
+   * Represents an object that can both compress and decompress messages.
+   */
+  public interface Codec extends Compressor, Decompressor {}
```

Comment:
Where is `Compressor` and `Decompressor` going? Only `Codec` is usable right now. When will the interfaces be used separately?